motion/thruster_interface_auv_cpp/include/thruster_interface_auv_cpp/thruster_interface_auv_driver.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef THRUSTER_INTERFACE_AUV_DRIVER_HPP
+#define THRUSTER_INTERFACE_AUV_DRIVER_HPP
+
+#include <vector>
+#include <string>
+#include <map>
+#include <iostream>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+#include <linux/i2c-dev.h>
+#include <cmath>
+
+class ThrusterInterfaceAUVDriver {
+public:
+
+    //default contructor called when instantiating the object in .hpp file
+    ThrusterInterfaceAUVDriver();
+    ~ThrusterInterfaceAUVDriver();
```

Comment:
yeah we dont actively use this in the code, it's there such that it compiles. This is because in the .hpp file for the node class i declare a driver class as field member of the node class and doing that it triggers a constructor, requesting me to create an "empty" constructor to handle this. Is there a better way to do it?

Reply:
Ah, you'd want to hold the ThrusterInterfaceAUVDriver object in a unique_ptr. That's the usual way of dealing with classes that need other (fairly big) class objects as part of their members. Especially in this case, where having a default-constructed ThrusterInterfaceAUVDriver is not "allowed" (in the sense that it will not be a valid object unless you make it with the appropriate constructor) So something like 

`std::unique_ptr<ThrusterInterfaceAUVDriver> thruster_interface_`

And then using `std::make_unique` where you define it. That will allow you to remove the default constructor.

Reply:
could you be more specific? What i did (and is not compiling with the same errors as when i hadn't the default constructor) is replacing

1. `ThrusterInterfaceAUVDriver thruster_driver_` with `std::unique_ptr<ThrusterInterfaceAUVDriver> thruster_driver_` as member of the class "blablabla_node"
2. removing the default constructor `ThrusterInterfaceAUVDriver();`
3. and replacing the line of code that instantiate the object passing the arguments with `this->thruster_driver_ = std::make_unique<ThrusterInterfaceAUVDriver>(arguments....`

The error is `error: no declaration matches ‘ThrusterInterfaceAUVDriver::ThrusterInterfaceAUVDriver()’
    3 | ThrusterInterfaceAUVDriver::ThrusterInterfaceAUVDriver() {}
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/alberto/vortex/auto_ws/src/vortex-auv/motion/thruster_interface_auv/src/thruster_interface_auv_driver.cpp:1:
/home/alberto/vortex/auto_ws/src/vortex-auv/motion/thruster_interface_auv/include/thruster_interface_auv/thruster_interface_auv_driver.hpp:30:7: note: candidates are: ‘ThrusterInterfaceAUVDriver::ThrusterInterfaceAUVDriver(const ThrusterInterfaceAUVDriver&)’
   30 | class ThrusterInterfaceAUVDriver {
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/alberto/vortex/auto_ws/src/vortex-auv/motion/thruster_interface_auv/include/thruster_interface_auv/thruster_interface_auv_driver.hpp:57:5: note:                 ‘ThrusterInterfaceAUVDriver::ThrusterInterfaceAUVDriver(short int, int, const std::vector<short int>&, const std::vector<short int>&, const std::vector<int>&, const std::vector<int>&, const std::vector<double>&, const std::vector<double>&)’
   57 |     ThrusterInterfaceAUVDriver(short I2C_BUS,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/alberto/vortex/auto_ws/src/vortex-auv/motion/thruster_interface_auv/include/thruster_interface_auv/thruster_interface_auv_driver.hpp:30:7: note: ‘class ThrusterInterfaceAUVDriver’ defined here
   30 | class ThrusterInterfaceAUVDriver {
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~
`

Reply:
update: fixed it